api/configuration/v1alpha1/kongpluginbinding_types.go

```diff
@@ -81,66 +93,55 @@ type KongPluginBindingSpec struct {
 	// 9. Consumer group
 	// 10. Route
 	// 11. Service
-	// 12. Global
 	//
-	// TODO(mlavacca): we need to figure out how to deal with global plugins. By means of this new API,
-	// KongClusterPlugin can be replaced by kongPluginBindings with no Kong references. This way we'd be
-	// more coherent with the Konnect approach.
-	// https://github.com/Kong/kubernetes-configuration/issues/7
-	Kong *KongReferences `json:"kong,omitempty"`
-
-	// TODO(mlavacca): let's defer this one to the future as we are not sure about the shape we want to give it.
-	// https://github.com/Kong/kubernetes-configuration/issues/8
-	// EntityReference        *GenericEntityRef `json:"genericEntityRef,omitempty"`
-}
+	// +optional
+	// +kubebuilder:validation:XValidation:message="group/kind not allowed for the routeRef",rule="(self.kind == 'KongRoute' && self.group == 'configuration.konghq.com') || (self.kind == 'Ingress' && self.group == 'networking.k8s.io') || (self.kind == 'HTTPRoute' && self.group == 'gateway.networking.k8s.io') || (self.kind == 'GCPRoute' && self.group == 'gateway.networking.k8s.io')"
+	RouteReference *TargetRefWithGroupKind `json:"routeRef,omitempty"`
 
-type KongReferences struct {
-	RouteReference         *EntityRef `json:"routeRef,omitempty"`
-	ServiceReference       *EntityRef `json:"serviceRef,omitempty"`
-	ConsumerReference      *EntityRef `json:"consumerRef,omitempty"`
-	ConsumerGroupReference *EntityRef `json:"consumerGroupRef,omitempty"`
+	// +optional
+	// +kubebuilder:validation:XValidation:message="group/kind not allowed for the serviceRef",rule="(self.kind == 'KongService' && self.group == 'configuration.konghq.com') || (self.kind == 'Service' && (self.group == '' || self.group == 'core'))"
+	ServiceReference       *TargetRefWithGroupKind `json:"serviceRef,omitempty"`
+	ConsumerReference      *TargetRef              `json:"consumerRef,omitempty"`
+	ConsumerGroupReference *TargetRef              `json:"consumerGroupRef,omitempty"`
```

Comment:
Can we add comments to those fields stating that they designate which respective entity to bind the plugin to?

Also I believe it might be worth adding comments to consumers and consumergroup refs to explicitly call out that only `KongConsumer` from `configuration.konghq.com` API group (and similarly for CG) is allowed and thus the ref name references objects of this type.

Reply:
Yep, I've added and improved the comments, see [86ac6fe](https://github.com/Kong/kubernetes-configuration/pull/11/commits/86ac6fe50f08c0a38fc7fe9e9626ff13a1f7ab5d)